homeassistant/components/sensor/onewire.py

```diff
@@ -0,0 +1,93 @@
+""" Support for DS18B20 One Wire Sensors"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELCIUS, TEMP_FAHRENHEIT
+from glob import glob
+import os
+import time
+import logging
+
+
+BASE_DIR = '/sys/bus/w1/devices/'
+DEVICE_FOLDERS = glob(os.path.join(BASE_DIR, '28*'))
+SENSOR_IDS = []
+DEVICE_FILES = []
+for device_folder in DEVICE_FOLDERS:
+    SENSOR_IDS.append(os.path.split(device_folder)[1])
+    DEVICE_FILES.append(os.path.join(device_folder, 'w1_slave'))
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the one wire Sensors"""
+
+    if DEVICE_FILES == []:
+        _LOGGER.error('No onewire sensor found.')
+        _LOGGER.error('Check if dtoverlay=w1-gpio,gpiopin=4.')
+        _LOGGER.error('is in your /boot/config.txt and')
+        _LOGGER.error('the correct gpiopin number is set.')
+        return
+
+    devs = []
+    names = SENSOR_IDS
+
+    for key in config.keys():
+        if key == "names":
+            # only one name given
+            if isinstance(config['names'], str):
+                names = [config['names']]
+            # map names and sensors in given order
+            elif isinstance(config['names'], list):
+                names = config['names']
+            # map names to ids.
+            elif isinstance(config['names'], dict):
+                names = []
+                for sensor_id in SENSOR_IDS:
+                    names.append(config['names'].get(sensor_id, sensor_id))
+    for device_file, name in zip(DEVICE_FILES, names):
+        devs.append(OneWire(name, device_file, TEMP_CELCIUS))
+    add_devices(devs)
+
+
+class OneWire(Entity):
+    """ A Dallas 1 Wire Sensor"""
+
+    def __init__(self, name, device_file, unit_of_measurement):
```

Comment:
The only value `unit_of_measurement` gets set to is `TEMP_CELCIUS` so it looks like `unit_of_measurement` can be removed as a parameter.